src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -121,4 +128,101 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  /**
+    * Returns the number of users currently recorded
+    */
+  public int getUserCount() {
+    return users.size();
+  }
+
+  /**
+    * Returns the users array
+    */
+  public List<User> getAllUsers(){
+    return users;
+  }
+
+  /**
+    * Returns the user with the most messages sent
+    * Pre : the the users list should already have the messageCount calculated
+    */
+  public String getTopUser(){
+    sortUserList();
+    return users.get(0).getName();
+  }
+
+  /**
+    * Iterate through all messages and get the count for the owner of ther messages
+    * Should only be called in PersistentDataStore when the messageCount attr does not exist
+    */
+  public int getMessageCountForUser(UUID uuid){
+    List<Message> messages = MessageStore.getInstance().getAllMessages();
+    int messageCount = 0;
+    for (Message message : messages) {
+      UUID id = message.getAuthorId();
+      if (id != null) {
+        if (uuid == id) {
+          messageCount++;
+        }
+      }
+    }
+    return messageCount;
+  }
+
+  /**
+    * Sort the users list by using sort in the library
+    */
+  public void sortUserList(){
+    Collections.sort(users, new Comparator<User>(){
+      public int compare(User a, User b){
+        return b.getMessageCount() - a.getMessageCount();
+      }
+    });
+  }
+
+  /**
+    * make a JSON file that consists JSON objects {user, messageCount}
+    * sample JSONFile :
+    * [
+    *   {messageCount : 20, name : userOne},
+    *   {messageCount : 10, name : userTwo}
+    * ]
+    */
+  public void writeJSON(){
+    // get the users array sorted with the corresponding messageCount setup
+    sortUserList();
+
+    try {
+      // check if the directory exists, if not, create it
+      File path = new File("./api/stats");
+      if (!path.isDirectory()){
+        path.mkdirs();
+      }
+      // make the JSON file in the directory ./api/stats
+      File file = new File("./api/stats/", "userData.json");
+      FileWriter fileWriter = new FileWriter(file);
+      int i = 1;  // fix for no counter in forEach loop, need to know when the loop ends to make it proper JSON file
+      fileWriter.write("[");
+      fileWriter.write("\n");
+      for (User user : users) {
+        JSONObject userObj = new JSONObject();
+        userObj.put("messageCount", user.getMessageCount());
+        userObj.put("name", user.getName());
+        fileWriter.write(userObj.toJSONString());
+        // adding "," between objects, don't add for the last object
```

Comment:
Consider simplifying this code.  For example, you can stick the userObj.toJSONStrings into an array and use String.Join to combine the list of objects.  

Can you also not add each userObj to an outer JsonObject and have this generated with outerJsonObj.toJsonString()?